alpakaConfig.cmake

```diff
@@ -623,6 +623,7 @@ LIST(APPEND _ALPAKA_LINK_LIBRARIES_PUBLIC "${_ALPAKA_LINK_LIBRARY}")
 
 # Add all the source and include files in all recursive subdirectories and group them accordingly.
 append_recursive_files_add_to_src_group("${_ALPAKA_SUFFIXED_INCLUDE_DIR}" "${_ALPAKA_SUFFIXED_INCLUDE_DIR}" "hpp" _ALPAKA_FILES_HEADER)
+append_recursive_files_add_to_src_group("${_ALPAKA_SUFFIXED_INCLUDE_DIR}" "${_ALPAKA_SUFFIXED_INCLUDE_DIR}" "h" _ALPAKA_FILES_HEADER)
```

Comment:
if we do this, we probably have to move the warning suppression into the `.h` header itself...

Reply:
We should make Alpaka an installable library now, so we can compile this to an internal object and making the header internal only ;-)

Reply:
I do not understand what adding this file to the IDE project has to do with the warnings and the installation. This line should not have any influence on compilation.

Reply:
Ah, now I see. The analysis build with the self-contained header check fails. You can remove the line and I will see if I can exclude it from this check later.

Reply:
Seen to late. Do you think the state with moving the suppression into the impl is fine as well?

Reply:
Btw: maybe I misinterpreted the first fail on clang and that was the same reason

Reply:
I liked the first version with unchanged header more. But both is good. You could try the clang suppression again.

Reply:
Me too, but more robust that way now.
Updated regarding clang now as you initially suggested.